api/src/lib/crud.js

```diff
@@ -0,0 +1,128 @@
+import koa from 'koa';
+import coBody from 'co-body';
+import koaRoute from 'koa-route';
+
+export default (queries, restricted, excludedRouteMethod) => {
+    const app = koa();
+    let q;
+
+    app.use(function* (next) {
+        q = queries(this.client);
+        yield next;
+    });
+
+    if (!excludedRouteMethod || excludedRouteMethod.indexOf('get') === -1) {
```

Comment:
I would remove this logical of excluded routes. Instead, we are using a middleware checking the method and throwing a 404 if it is not allowed. For instance, something like (not tested code):

```
app.use(function* (next) {
        if (this.method !== 'GET') {
            this.status = 404;
            return;
        }

        yield next;
    });
```

Which is more readable and more flexible.